test/system/src/main/java/io/pravega/test/system/framework/services/docker/BookkeeperDockerService.java

```diff
@@ -0,0 +1,103 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.test.system.framework.services.docker;
+
+import com.spotify.docker.client.messages.ContainerConfig;
+import com.spotify.docker.client.messages.mount.Mount;
+import com.spotify.docker.client.messages.swarm.ContainerSpec;
+import com.spotify.docker.client.messages.swarm.EndpointSpec;
+import com.spotify.docker.client.messages.swarm.NetworkAttachmentConfig;
+import com.spotify.docker.client.messages.swarm.PortConfig;
+import com.spotify.docker.client.messages.swarm.ResourceRequirements;
+import com.spotify.docker.client.messages.swarm.Resources;
+import com.spotify.docker.client.messages.swarm.RestartPolicy;
+import com.spotify.docker.client.messages.swarm.ServiceMode;
+import com.spotify.docker.client.messages.swarm.ServiceSpec;
+import com.spotify.docker.client.messages.swarm.TaskSpec;
+import java.net.URI;
+import lombok.extern.slf4j.Slf4j;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import static io.pravega.test.system.framework.Utils.DOCKER_NETWORK;
+
+@Slf4j
+public class BookkeeperDockerService extends DockerBasedService {
+
+    private static final int BK_PORT = 3181;
+    private final long instances = 3;
+    private final double cpu = 0.1;
+    private final double mem = 1024.0;
+    private final URI zkUri;
+
+    public BookkeeperDockerService(String serviceName, final URI zkUri) {
+        super(serviceName);
+        this.zkUri = zkUri;
+    }
+
+    public void stop() {
+        super.stop();
+    }
+
+    @Override
+    public void clean() {
+    }
+
+    @Override
+    public void start(final boolean wait) {
+        start(wait, setServiceSpec());
+    }
+
+    private ServiceSpec setServiceSpec() {
+
+        Map<String, String> labels = new HashMap<>();
+        labels.put("com.docker.swarm.service.name", serviceName);
+
+        Mount mount1 = Mount.builder().type("volume").source("index-volume").target("/bk/index")
+                .build();
+        Mount mount2 = Mount.builder().type("volume").source("logs-volume")
+                .target("/opt/dl_all/distributedlog-service/logs/")
+                .build();
+        //TODO: add journal and ledger mounted volumes
+        String zk = zkUri.getHost() + ":" + ZKSERVICE_ZKPORT;
+        List<String> stringList = new ArrayList<>();
+        String env1 = "ZK_URL=" + zk;
+        String env2 = "ZK=" + zk;
+        String env3 = "bookiePort=" + String.valueOf(BK_PORT);
+        String env4 = "DLOG_EXTRA_OPTS=-Xms512m";
+        stringList.add(env1);
+        stringList.add(env2);
+        stringList.add(env3);
+        stringList.add(env4);
+        final TaskSpec taskSpec = TaskSpec
+                .builder().restartPolicy(RestartPolicy.builder().maxAttempts(0).condition("none").build())
```

Comment:
If I'm reading this right, then we aren't retrying here, is this the expected behavior?

Reply:
Yes, if we enable this. If there is a problem with container, it will still keep on restarting and we might never know that it has some issue. This ensures that the container fails/exits with a status other than `0`, if there is an issue with container deployment with that config.

Reply:
Ok.